### ISSUE TITLE
Allow specifying locale for `react/jsx-sort-props`

### ISSUE THREAD

**Issue by Hazeman28** [yannickcr/eslint-plugin-react#3002]
We work in a team where people have systems with different locales. When we add `react/jsx-sort-props` to our `eslint` config, it behaves differently for each system.

For example, on a `en-US` system, props for the component below are sorted like this:
```jsx
<RawFileField
  onChange={handleChange}
  onFileRemove={asMedia ? null : handleRemove}
  {...props}
/>
```
But on a `sk-SK` system, they're sorted like this:
```jsx
<RawFileField
  onFileRemove={asMedia ? null : handleRemove}
  onChange={handleChange}
  {...props}
/>
```

This results in `eslint` tests passing locally, but failing during CI/CD pipelines. Ability to specify which locale to use for sorting would be nice.

**Comment by ljharb** [comment]
That seems like a reasonable option - it would default to the current locale, and you could specify whatever value you wanted in the config.

**Comment by ljharb** [comment]
Note that in node < 13, the `sk-SK` locale doesn't exist, so we should probably add some test cases for a locale that sorts differently than en-US, that's present in older versions of node.
